#163396359 delete party

Hoxtygen opened the pull request:
#### What does this PR do?
ensure a given party can be deleted

#### Description of Task to be completed?
Have the following endpoint working
`DELETE /api/v1/parties/<party-id>`

#### How should this be manually tested?
clone the repo and cd into it then on the terminal do `npm run dev`
Using postman, test the endpoint with the following header information
- key: Content-Type value: application/json


#### What are the relevant pivotal tracker stories?
 [#163396359](https://www.pivotaltracker.com/story/show/163396359)

#### Screenshots (if appropriate)
![deleteparty](https://user-images.githubusercontent.com/19408669/51696308-7c122500-2005-11e9-8570-5685d4489dc7.PNG)

Hoxtygen committed:
ft(feature): Delete a party

ensure a given party can be deleted

[Starts ft-delete-party-#163396359]

Hoxtygen committed:
ft(feature): Delete a party

ensure a given party can be deleted

[Delivers ft-delete-party-#163396359]

Orlayhemmy commented on server/controllers/partiesController.js:
```diff
@@ -22,6 +22,27 @@ class partyController {
+  static deleteParty(req, res) {
+    const id = parseInt(req.params.id, 10);
+    let deleted;
+    Parties.map((party, index) => {
```
Your implementation here needs to be changed for efficiency. Instead of mapping, since the list of party is an array, why not do a filter and return a response immediately. I would advice you go and read about array filter

Hoxtygen replied:
Okay, the filter method has been implemented.

Hoxtygen committed:
refactor: use array filter method

Change array.prototype.map() method to array.prototype.filter() method
to remove elements from the database. This ensures effieciency in that
the whole of the array is not looped through to get to the element to be
deleted.

[Delivers #163396359]
